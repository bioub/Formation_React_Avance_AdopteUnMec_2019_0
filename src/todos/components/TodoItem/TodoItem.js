import React from 'react';
import css from './TodoItem.module.scss';
import { shape, string, number, bool } from 'prop-types';
import { hideable } from '../../../hoc/hideable';

export function TodoItem({ item, onDeleteItem = () => {} }) {
  return (
    <div className={css.TodoItem}>
      <span>{item.text}</span>
      <button onClick={() => onDeleteItem(item)}>-</button>
    </div>
  );
}

TodoItem.propTypes = {
  item: shape({
    id: number,
    text: string.isRequired,
    completed: bool,
  }),
};

export const HideableTodoItem = hideable(TodoItem);
