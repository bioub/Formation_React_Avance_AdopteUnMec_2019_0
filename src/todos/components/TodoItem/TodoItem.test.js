import renderer from 'react-test-renderer';
import React from 'react';
import { TodoItem } from './TodoItem';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const todo = {
    id: 123,
    text: 'Acheter du pain',
    completed: false,
  };
  const json = renderer.create(<TodoItem item={todo} />).toJSON();
  expect(json).toMatchSnapshot();
});

it('should contains span with todo.text', () => {
  const todo = {
    id: 123,
    text: 'Acheter du pain',
    completed: false,
  };
  const wrapper = shallow(<TodoItem item={todo} />);

  expect(wrapper.find('span').text()).toBe('Acheter du pain');
});

it('should call onDeleteItem on click', () => {
  const todo = {
    id: 123,
    text: 'Acheter du pain',
    completed: false,
  };
  const mock = jest.fn();
  const wrapper = shallow(<TodoItem item={todo} onDeleteItem={mock} />);

  wrapper.find('button').simulate('click');

  expect(mock).toHaveBeenNthCalledWith(1, todo);
});
