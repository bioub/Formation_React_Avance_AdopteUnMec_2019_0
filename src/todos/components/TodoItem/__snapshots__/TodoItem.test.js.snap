// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="TodoItem"
>
  <span>
    Acheter du pain
  </span>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
